### PR TITLE
Fixed a problem with Git Stage Filepath

### PR DESCRIPTION
**Pull request by jpcercal**
Dear @sjparkinson,

I wrote a fix to your code.

The path before this fix: string(63) "C:/xampp/htdocs/silex-scaffolding\bin/git/hooks/pre-commit.php
"

The same path before this fix: string(63) "C:/xampp/htdocs/silex-scaffolding\bin/git/hooks/pre-commit.php"

### CODE REVIEW

File: src/VersionControl/GitVersionControl.php [hunk 1]
```diff
@@ -42,7 +42,7 @@ public function getStagedFiles()
         foreach ($this->getFiles() as $file) {
             list($status, $relativePath) = explode("\t", $file);
 
-            $fullPath = $base . DIRECTORY_SEPARATOR . $relativePath;
+            $fullPath = trim($base . DIRECTORY_SEPARATOR . $relativePath);
```
Comment (keradus) [inline src/VersionControl/GitVersionControl.php]: rtrim ?

**Comment by keradus** [comment]
It would be great to have tests to test the bug and fix for it

**Comment by sjparkinson** [comment]
Thanks for your changes! I've merged them in https://github.com/sjparkinson/static-review/commit/1c2d33b3de927c3e34895a84ef0ea4a4dcc99e57.

File: src/VersionControl/GitVersionControl.php (hunk 1 above)
Comment (keradus) [inline src/VersionControl/GitVersionControl.php]: :(

Comment (sjparkinson) [inline src/VersionControl/GitVersionControl.php]: Feel free to open a new PR with the change. I saw no reason for this to hold up a merge though.

It'd be helpful to explain why `rtrim` would be better than `trim`.

Comment (keradus) [inline src/VersionControl/GitVersionControl.php]: trim may be described as rtirm+ltrim, and here you don't need ltrim, so why to waste time on doing it?

Comment (sjparkinson) [inline src/VersionControl/GitVersionControl.php]: > trim may be described as rtirm+ltrim, and here you don't need ltrim, so why to waste time on doing it?

Thanks, that's a much more helpful comment, but I'll reiterate my statement about this micro-optimisation not being a concern enough to hold up the merge. Like I said though, I'd happily review a new PR for the change.

Comment (keradus) [inline src/VersionControl/GitVersionControl.php]: I understand that.
And still, I may only guess that noone will send it ;)

**Pull request closed without merging**